katello/rubygem-katello/rubygem-katello.spec

```diff
@@ -0,0 +1,1788 @@
+BuildRequires: %{?scl_prefix}rubygem(bastion) >= 6.0.0
+BuildRequires: %{?scl_prefix}rubygem(bastion) < 7.0.0
+BuildRequires: %{?scl_prefix}rubygem(foreman-tasks) >= 0.8.0
+BuildRequires: %{?scl_prefix}rubygem(foreman-tasks) < 1.0.0
+BuildRequires: %{?scl_prefix}rubygem(gettext_i18n_rails)
+BuildRequires: %{?scl_prefix}rubygem(apipie-rails) >= 0.1.1
+BuildRequires: %{?scl_prefix}rubygem(runcible) >= 2.0.0
+BuildRequires: %{?scl_prefix}rubygem(anemone)
+BuildRequires: %{?scl_prefix}rubygem(deface) >= 1.0.0
+BuildRequires: %{?scl_prefix}rubygem(deface) < 2.0.0
+BuildRequires: %{?scl_prefix}rubygem(qpid_messaging)
+BuildRequires: %{?scl_prefix_ror}rubygem(rails)
+BuildRequires: %{?scl_prefix_ruby}rubygem(json)
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}ruby(rubygems)
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+
+#old dependencies that are no longer needed, remove in katello 2.6
+Obsoletes: ruby193-rubygem-haml
+Obsoletes: ruby193-rubygem-haml-rails
+Obsoletes: ruby193-rubygem-hpricot
+Obsoletes: ruby193-rubygem-strong_parameters
+Obsoletes: ruby193-rubygem-tire
+Obsoletes: ruby193-rubygem-less
+Obsoletes: ruby193-rubygem-hashr
+Obsoletes: ruby193-rubygem-foreman_gutterball
+
+BuildArch: noarch
+Provides: %{?scl_prefix}rubygem(katello) = %{version}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}}
+
+%description
+Katello
+
+%package doc
+BuildArch:  noarch
+Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+Summary:    Documentation for rubygem-%{gem_name}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}-doc}
+
+%description doc
+This package contains documentation for rubygem-%{gem_name}.
+
+%package -n %{katello_ostree}
+Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+Summary:    Katello Ostree Plugin
+
+%description -n %{katello_ostree}
+This package provides the ostree plugin for rubygem-%{gem_name}.
+
+%prep
+%setup -n %{pkg_name}-%{version} -q -c -T
+%{?scl:scl enable %{scl} - <<EOF}
+%gem_install -n %{SOURCE0}
+%{?scl:EOF}
+
+%build
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+%foreman_bundlerd_file
+%foreman_precompile_plugin -s -a
+
+%files
+%dir %{gem_instdir}/
+%{gem_libdir}
+%{gem_instdir}/app
+%{gem_instdir}/config
+%{gem_instdir}/db
+%{gem_instdir}/locale
+%{gem_instdir}/engines
+%{gem_instdir}/ca
+%{gem_instdir}/vendor
+%{gem_spec}
+%{foreman_bundlerd_plugin}
+%{foreman_apipie_cache_foreman}
+%{foreman_apipie_cache_plugin}
+%{foreman_assets_plugin}
+%{gem_instdir}/public/assets/bastion_katello
+
+%exclude %{gem_cache}
+%exclude %{gem_instdir}/lib/katello/repository_types/ostree.rb
+
+%doc %{gem_instdir}/LICENSE.txt
+
+%files doc
+%doc %{gem_docdir}
+%doc %{gem_instdir}/README.md
+
+%files -n %{katello_ostree}
+%{gem_instdir}/lib/katello/repository_types/ostree.rb
+
+%changelog
+* Wed Jul 29 2015 Eric D. Helms <ericdhelms@gmail.com> 2.4.0-1.nightly
```

Comment:
Looks like this wasn't updated in a while

Reply:
That's cause it builds only for nightlies and release branches. I'm not sure there is value in updating it given that.

Reply:
It might make sense to import some changes from released branches. I think right now 3.5 will not have the 3.4 changelog either.

Reply:
That's fairly true, at a minimum this could follow the Foreman procedure which just adds a line for each major version bump. Not adding a lot of value, but I suppose then you'd at least see this go up over time. These nightly packages are special to the point of even if we had the changelog from each release branch, it would not truly capture the changes to the spec over time. Perhaps the answer is everytime we modify the spec, we need to bump and "release" it to koji as a tagged and built version.

Reply:
Interesting idea. I do wonder how many people actually look at it vs the git log and with the branching that's going on ... Would it make sense to just drop the changelog?

Reply:
I think its good RPM practice, and on a release branch it adds value for those who know RPM. I'd recommend we just set this every time we branch. I'll update it for this PR.